pkg/controller/plan/kubevirt.go

```diff
@@ -555,11 +573,131 @@ func (r *KubeVirt) getPVCs(vm *plan.VMStatus) (pvcs []core.PersistentVolumeClaim
 		pvcAnn := pvc.GetAnnotations()
 		if pvcAnn[kVM] == vmLabels[kVM] && pvcAnn[kPlan] == vmLabels[kPlan] {
 			pvcs = append(pvcs, *pvc)
+		} else {
+			if r.isOvirtProvider(vm) {
+				ovirtVm := &ovirt.Workload{}
+				err = r.Source.Inventory.Find(ovirtVm, vm.Ref)
+				if err != nil {
+					return
+				}
+				for _, da := range ovirtVm.DiskAttachments {
+					if da.Disk.ID == pvc.Spec.DataSource.Name {
+						pvcs = append(pvcs, *pvc)
+						break
+					}
+				}
+			}
+		}
+	}
+	return
+}
+
+func (r *KubeVirt) createVolumes(vm ref.Ref) (err error) {
+	ovirtVm := &ovirt.Workload{}
+	err = r.Source.Inventory.Find(ovirtVm, vm)
+	if err != nil {
+		return
+	}
+	sourceUrl, err := url.Parse(r.Source.Provider.Spec.URL)
+	if err != nil {
+		return
+	}
+
+	storageName := &r.Context.Map.Storage.Spec.Map[0].Destination.StorageClass
+	for _, da := range ovirtVm.DiskAttachments {
+		populatorCr := r.OvirtVolumePopulator(da, sourceUrl)
+		failure := r.Client.Create(context.Background(), populatorCr, &client.CreateOptions{})
+		if failure != nil && !k8serr.IsAlreadyExists(failure) {
+			return failure
+		}
+
+		accessModes, volumeMode, failure := r.getStorageProfileModes(*storageName)
+		if failure != nil {
+			return failure
+		}
+
+		pvc := r.Builder.PersistentVolumeClaimWithSourceRef(da, storageName, populatorCr.Name, accessModes, volumeMode)
```

Comment:
Check for `nil` would be useful, although it cannot happen with current state of the code.